### ISSUE TITLE
[HOLD for payment 2024-10-14] [$250] Reopened my laptop and saw Uh oh, something's wrong!

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#49340]
`If` you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** Needs Reproduction
**Reproducible in production?:** Needs Reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1726529947176339

## Action Performed:
1. Leave the laptop in sleep/hibernation mode with  www.staging.new.expensify.com open
2. Reopen the laptop after it was in sleep/hibernation mode.
## Expected Result:
The staging environment should load normally without any error messages.
## Actual Result:
The site displays the "Uh oh, something's wrong!" error upon access.
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
Logs

[Logs 1.txt](https://github.com/user-attachments/files/17029284/Logs.1.txt)

[Logs 2.txt](https://github.com/user-attachments/files/17029298/Logs.2.txt)

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021838649543194312084</li>
        <li>Upwork Job ID: 1838649543194312084</li>
        <li>Last Price Increase: 2024-10-01</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Christinadobrzyn</details>

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by Christinadobrzyn** [comment]
I'm not going to be able to get to this today - I'll test tomorrow and let you know next steps @rezkiy37!

**Comment by rezkiy37** [comment]
We are internally discussing a solution to improve the page refresh.

**Comment by rezkiy37** [comment]
Hello!
I am OOO from 19.09 - 23.09. I will continue to work on this one when I am back.

**Comment by Christinadobrzyn** [comment]
@rezkiy37 is coming back in a few days - they'll continue the work then

**Comment by rezkiy37** [comment]
I am back 🙂

**Comment by rezkiy37** [comment]
I am preparing a proposal.

**Comment by Christinadobrzyn** [comment]
Side note, I'm going to be ooo until 9/30. I don't think we'll need a BZ teammate on this since we're working/reviewing proposals. So I'm just going to keep myself assigned. If you need someone, please reach out to the team in Slack for a volunteer. TY!

**Comment by DylanDylann** [comment]
I need some time to verify the solution. If we don't rush, I will complete in the next week

**Comment by Christinadobrzyn** [comment]
Update for Mevlin- we're working on a solution

**Comment by DylanDylann** [comment]
@rezkiy37 I can't reproduce this bug. It seems to make sense with your explanation that this bug only happens if while the laptop is sleeping we have a new deployment.

> A user may have the application open in a browser tab during a new deployment. If so, the required JS file may no longer be available on your server. So they will receive an error when the browser tries to load the file.

How can we verify this point? Is there any way to simulate a new deployment?

**Comment by rezkiy37** [comment]
@DylanDylann, no way to simlulate. It should be a real deployment.

**Comment by DylanDylann** [comment]
@rezkiy37 How can we verify your RCA is correct? It is only your assumption, right?

**Comment by Christinadobrzyn** [comment]
based on - https://github.com/Expensify/App/issues/49340#issuecomment-2385219458

It sounds like we might need to close this as Needs Reproduction. Also, this is part of the vip-vsb project which has been tabled so I think we can close this without action but let me know if you feel otherwise!

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by rezkiy37** [comment]
> @rezkiy37 How can we verify your RCA is correct? It is only your assumption, right?

@DylanDylann, it is not an assumption. There is a pretty tricky bug and it is not such easy to reproduce it. However, we have the attached log files. Where described the error. We found a solution that allows the user to get the app working again after our internal investigations and research. I described it in the [proposal](https://github.com/Expensify/App/issues/49340#issuecomment-2374458620). Moreover, this improvement enhances the app refresh after some сaught crashes. So I kindly ask you to consider the proposal as a real improvement and fix this bug.

cc @Christinadobrzyn

**Comment by DylanDylann** [comment]
@rezkiy37 Wait a moment.

**Comment by rezkiy37** [comment]
Btw, this is a previous issue (https://github.com/Expensify/App/issues/44859) and PR (https://github.com/Expensify/App/pull/45596) for this.

**Comment by rezkiy37** [comment]
There are test steps to simulate it if edit code:

> For the scenario when a chunk is failed and we need to update the page:
> 1. Run the web app
> 2. Open dev tools and choose the application -> session storage
> 3. Observe RETRY_LAZY_REFRESHED variable
> 4. Open lazyRetry.ts and add  const _componentImport = () => (hasRefreshed ? componentImport() : > Promise.reject(componentImport)); then replace componentImport().then(... to _componentImport().then(...
> 5. Observe that the page has refreshed, RETRY_LAZY_REFRESHED has changed to true and then to false

**Comment by DylanDylann** [comment]
We are using https://github.com/bvaughn/react-error-boundary/releases/tag/4.0.11 

Currently, in our App, we call `resetBoundary` function (this is a built-in function). let's see this function internally

<img width="834" alt="Screenshot 2024-10-02 at 18 01 48" src="https://github.com/user-attachments/assets/50c39fd9-83c2-43a0-bcf0-58f61285a8bb">


We are calling `resetErrorBoundary` function. And let's see this code 

<img width="1094" alt="Screenshot 2024-10-02 at 18 03 39" src="https://github.com/user-attachments/assets/1c31596e-0627-4644-9b11-a38aba52d63d">

resetErrorBoundary function already retried the render

@rezkiy37  But as I understand, we need to reload the page to fix this issue instead of only re-rendering.


Another thing resetErrorBoundary also called `props.onReset`

<img width="1096" alt="Screenshot 2024-10-02 at 18 07 45" src="https://github.com/user-attachments/assets/2aa8e5e5-ff55-4eee-bd04-f6df058957f3">

@rezkiy37 What do you think if we solve this problem by passing onReset to ErrorBoundary Component

**Comment by DylanDylann** [comment]
> @DylanDylann, it is not an assumption. There is a pretty tricky bug and it is not such easy to reproduce it. However, we have the attached log files. Where described the error. We found a solution that allows the user to get the app working again after our internal investigations and research. I described it in the https://github.com/Expensify/App/issues/49340#issuecomment-2374458620. Moreover, this improvement enhances the app refresh after some сaught crashes. So I kindly ask you to consider the proposal as a real improvement and fix this bug.


Agreed 🟢 🟢 🟢

**Comment by rezkiy37** [comment]
> @rezkiy37 What do you think if we solve this problem by passing onReset to ErrorBoundary Component

This bug is platform-specific and cannot happen on native platforms (_IOS and Android_). So we need to apply it properly. I have a draft PR (https://github.com/Expensify/App/pull/49341) where I implemented it. So you can take a look at this. In short, my idea is to use a hook (`usePageRefresh`) with a simple refresh for native and refresh + reload for web and desktop.

**Comment by DylanDylann** [comment]
@rezkiy37 are your PR ready?

**Comment by DylanDylann** [comment]
I already checked your PR. In this comment https://github.com/Expensify/App/issues/49340#issuecomment-2388382455, I think in the [react-error-boundary lib](https://github.com/bvaughn/react-error-boundary/releases/tag/4.0.11) they already exposed onReset props for us to customize the logic. For me, using this prop will be better. Wdyt?

**Comment by rezkiy37** [comment]
> @rezkiy37 are your PR ready?

Yes, almost done.

**Comment by rezkiy37** [comment]
> I already checked your PR. In this comment [#49340 (comment)](https://github.com/Expensify/App/issues/49340#issuecomment-2388382455), In the [react-error-boundary lib](https://github.com/bvaughn/react-error-boundary/releases/tag/4.0.11) they already exposed onReset props for us to customize the logic. So I think using this prop will be better. Wdyt?

Yeah, I got your idea. However, we still need to split it up by platforms. So we will utilize some hook or function and create `index.native.ts` and `index.ts`. Moreover, we will apply it implicitly, because the app calls reset explicitly and a rerender is expected, but `onReset` needs to be declared in another file. I mean the app calls `resetBoundary` and it reloads the page as well which is weird because the app does not call `reloadPage` `in GenericErrorPage.ts`. It sounds like a side effect. That's my opinion. WDYT, @DylanDylann?

**Comment by DylanDylann** [comment]
Yeah, 2 ways are similar about the implementation. No problem if we continue on your way, let's open PR

**Comment by rezkiy37** [comment]
Cool, I will do it today 🙂

**Comment by rezkiy37** [comment]
> Cool, I will do it today 🙂

Tomorrow.

**Comment by Christinadobrzyn** [comment]
@DylanDylann would you mind accepting the offer - https://www.upwork.com/nx/wm/offer/104422906

Also, do we need a regression test for this?

**Comment by Christinadobrzyn** [comment]
Paid out based on this payment summary - https://github.com/Expensify/App/issues/49340#issuecomment-2411922839

@DylanDylann can you let us know about a regression test?

**Comment by DylanDylann** [comment]
@Christinadobrzyn I am on it

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **NA**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: This seems like a new improvement
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


Regression Test Proposal

1. Leave the laptop in sleep/hibernation mode with [www.staging.new.expensify.com](http://www.staging.new.expensify.com/) open
2. Wait until there is a new deployment
3. Reopen the laptop after it was in sleep/hibernation mode.
4. Verify that the App is reloaded when clicking the second time on the refresh button 



Do we agree 👍 or 👎

**Comment by Christinadobrzyn** [comment]
Awesome! Thanks @DylanDylann! 

Regression test - https://github.com/Expensify/Expensify/issues/436834

I think we're good to close this!
